Fix:
error: implicit declaration of function 'getpagesize' is invalid in C99
on Apple Silicon systems.
https://jira.mongodb.org/browse/CDRIVER-4505
https://github.com/mongodb/mongo-c-driver/commit/a73297b3402c2acfdf47895b8d657b369fae7a38
--- src/libmongoc/src/mongoc/mongoc-counters.c.orig
+++ src/libmongoc/src/mongoc/mongoc-counters.c
@@ -24,6 +24,7 @@
 #ifdef BSON_OS_UNIX
 #include <sys/mman.h>
 #include <sys/shm.h>
+#include <unistd.h>
 #endif
 
 #ifdef _MSC_VER
@@ -111,7 +112,7 @@ mongoc_counters_calc_size (void)
            (n_cpu * n_groups * sizeof (mongoc_counter_slots_t)));
 
 #ifdef BSON_OS_UNIX
-   return BSON_MAX (getpagesize (), size);
+   return BSON_MAX (sysconf (_SC_PAGESIZE), size);
 #else
    return size;
 #endif
